4_advanced_usage/47_node_properties_data_sources.txt

```diff
@@ -0,0 +1,71 @@
+=== Node properties data source
+
+As explain in previous chapters, Nodes have properties that can be
+used to create groups or in techniques and directives parameters.
+These properties are key/value pairs, with value being a simple
+string or a well formed JSON document. The simplest way to define a
+property for a Node is thanks Rudder REST API.
+
+Rudder 4.1 introduces a new way to define Node properties by defining
+data sources.
+
+The following schema explains the general working process of data source:
+
+image:./images/rudder-datasources-description.png[Data source description]
+
+A displayed, a data source provides a way for Rudder to query (when some
+condition are met) a third party REST API to retrieve interesting
+JSON properties for nodes and save them for a given Node property key.
+
+More preciselly, there is three main set of properties to define (by UI
+or via Rudder API) to configure a data source:
+
+==== First set: data source description
+
+The first set of properties allows to define an unique identifier for
+the data source, which will be used as the property key in node, along
+with a human readable name and description.
+
+==== Second set: query configuration
+
+The second set of properties allow to define how the third party REST API will
+be queried and the returned JSON response processed.
+
+For now, we only support one query mode which is to do one HTTP query for each
+node. In the future, a mode where only one query is done to retrieve
+information for all nodes will be added.
+
+For the query, you will define the HTTP method to use (GET or POST), on what is the
+remote URL, if there is specific header or query parameter to add.
+
+If no answer or an error is returned for a node, the corresponding node
+property is deleted.
+
+When a JSON document is returned, you can define a JSON path expression
+(cf https://github.com/jayway/JsonPath/) to select only a sub-part of
+the document as the actual data to use as a node property.
+
+Finally, the resulting data is assigned to the node, using the data source
+unique identifier as property key.
+
+You can use Rudder variables expansion (`${ruder.node.xxx}`,
+`${rudder.parameter.xxx}`, `${node.properties[xxx]}`) in most of these
+configuration option: URL, headers, query parameters, JSON path.
```

Comment:
Please add "They will be replaced by their values for each node at the time the HTTP query is run."